core/transaction_pool/CMakeLists.txt

```diff
@@ -3,9 +3,21 @@
 # SPDX-License-Identifier: Apache-2.0
 #
 
+add_subdirectory(impl)
+
 add_library(transaction_pool_error
     transaction_pool_error.cpp
     )
 target_link_libraries(transaction_pool_error
     outcome
     )
+
+add_library(transaction_pool
```

Comment:
You should either define `pool_moderator` in this CMakeLists file and not include `impl` or define `transaction_pool` in the `impl/CMakeLists`, but not both

Reply:
Sound reasonable.